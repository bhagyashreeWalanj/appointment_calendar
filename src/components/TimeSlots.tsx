import React, { useState, useEffect } from 'react'
import { getAllSlotsWithStatus } from '../helpers/dataParser'
import styles from '../styles/TimeSlots.module.scss'
import { format } from 'date-fns'
import Notification from './Notification'
import { status } from '../actions/constants'
import { ToastContainer } from 'react-toastify'

interface ITimeSlots {
  selectedDate: Date
  slots: any
  mentor: string
  scheduleCallFn(time: string, date: string, status: string): void
}

const TimeSlots = ({
  selectedDate,
  slots,
  mentor,
  scheduleCallFn,
}: ITimeSlots) => {
  const [timeSlots, setTimeSlots] = useState([])
  const [submittedDate, setsubmittedDate] = useState(selectedDate)
  const [reason, setReason] = useState('')

  useEffect(() => {
    if (Object.keys(slots).length !== 0) {
      const year = selectedDate.getFullYear()
      const month = selectedDate.getMonth()
      const date = selectedDate.getDate()

      const slotsInHrsArray = slots[year]
        ? slots[year][month]
          ? slots[year][month][date]
            ? slots[year][month][date]
            : []
          : []
        : []
      const timingArray = getAllSlotsWithStatus(slotsInHrsArray, selectedDate)
      setTimeSlots(timingArray)
    }
  }, [slots, selectedDate, submittedDate])

  const handleOnChange = (e: any) => {
    setReason(e.target.value)
  }

  const handleSubmit = (selectedSlot: any, id: string) => {
    if (reason) {
      scheduleCallFn(
        selectedSlot.hour,
        selectedDate.getDate().toString(),
        'BOOKED',
      )
      Notification(
        'success',
        `You are scheduled a call with ${mentor} on Day ${format(
          selectedDate,
          'PPPP',
        )} 
      at ${selectedSlot.timeSlot} with following message: ${reason}`,
        setReason(''),
      )
      saveBookedSlot(selectedSlot)
      setsubmittedDate(selectedSlot.hour)
      document.getElementById(id)?.click()
    } else {
      Notification('warning', `You haven't added any reason`)
    }
  }
  const saveBookedSlot = (selectedSlot: any) => {
    const date = selectedDate.getDate().toString()
    const existingDate = localStorage.getItem(date)
    const time = selectedSlot.timeSlot.split(':')[0]
    localStorage.setItem(date, existingDate ? existingDate + ',' + time : time)
  }

  const handleOnSlotClick = (currentSlotStatus: string) => {
    if (currentSlotStatus !== status.AVAILABLE) {
      Notification(
        'error',
        `Slot is already booked. Please check available slots`,
      )
    }
  }

  return (
    <>
      <ToastContainer />
      <div className={styles.timeslots}>
        <div className={styles.timeslots__header}>
          <h4>{format(selectedDate, 'PPPP')}</h4>
        </div>
        <div className={styles.timeslots__body}>
          <input
            type="radio"
            name="openCloseAccordion"
            id="acc-close"
            className={styles.hide_button}
          />
          <div className={styles.tabs}>
            {timeSlots.map((slot: any, index: number) => {
              return (
                <div
                  className={styles.tab}
                  key={slot.timeSlot}
                  onClick={() => handleOnSlotClick(slot.status)}
                >
                  <input
                    id={`checkbox-${index}`}
                    type="radio"
                    name="openCloseAccordion"
                    style={{ display: 'none' }}
                  />

                  <label
                    className={
                      slot.status === status.AVAILABLE
                        ? styles.tab_label
                        : slot.status === status.NOT_AVAILABLE
                        ? styles.allotted
                        : styles.booked
                    }
                    id={`acc-open-${index}`}
                    htmlFor={
                      slot.status !== status.AVAILABLE
                        ? ''
                        : `checkbox-${index}`
                    }
                  >
                    <span>{slot.timeSlot}</span>
                    {slot.status}
                  </label>
                  <div className={styles.tab_content}>
                    <h4>Reason for a Call :</h4>
                    <input
                      type="text"
                      name={`message_${index}`}
                      className={styles.textBox}
                      id={`message-${index}`}
                      value={reason}
                      onChange={(e) => handleOnChange(e)}
                    />
                    <button
                      id={`submit-${index}`}
                      type="submit"
                      className={styles.confirmButton}
                      onClick={() => handleSubmit(slot, `close_${index}`)}
                    >
                      Confirm Call
                    </button>
                  </div>
                  <label
                    className={styles.box_close}
                    id={`close_${index}`}
                    htmlFor="acc-close"
                  ></label>
                </div>
              )
            })}
          </div>
        </div>
      </div>
    </>
  )
}

export default TimeSlots
