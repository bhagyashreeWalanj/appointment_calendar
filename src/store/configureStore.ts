import { configureStore } from '@reduxjs/toolkit'
import logger from "redux-logger";
import createSagaMiddleware from 'redux-saga'
import rootReducer, { IAppState } from '../reducers/rootReducer'
import { Store } from 'redux'
import {rootSaga} from '../sagas/rootSaga'

const sagaMiddleware = createSagaMiddleware()

export default function configuredstore(): Store<IAppState, any> {
  const store = configureStore({ reducer: rootReducer,
          middleware: (getDefaultMiddleware) => getDefaultMiddleware({
              thunk : false
          })
          .concat(sagaMiddleware).concat(logger)});
  sagaMiddleware.run(rootSaga)
  return store
}


