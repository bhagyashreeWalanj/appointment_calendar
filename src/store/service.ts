import axios from 'axios';

export const AXIOS_URL = 'https://private-anon-3947622ccf-cfcalendar.apiary-mock.com/mentors/1/agenda'

export const fetchSlots  = () => axios.get(AXIOS_URL);

