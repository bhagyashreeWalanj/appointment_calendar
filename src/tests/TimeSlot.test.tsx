import React from 'react'
import { shallow } from 'enzyme'
import TimeSlots from '../components/TimeSlots'
import { formattedSlots } from '../store/mockdata'
import { withHooks } from 'jest-react-hooks-shallow'

describe('Testing Time Slot component', () => {
  const scheduleCallFn = jest.fn()
  const timeSlotList = formattedSlots
  const selectedDate = new Date(2022, 7, 23) // 2022-07-24

  withHooks(() => {
    const wrapper = shallow(
      <TimeSlots
        selectedDate={selectedDate}
        slots={timeSlotList}
        mentor={'bhagyashree'}
        scheduleCallFn={() => scheduleCallFn(12, 23, 'BOOKED')}
      />,
    )

    it('should display to Time Slot Component', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should display Selected Date in Header', () => {
      const headerDate = wrapper.find('.timeslots__header')
      expect(headerDate.html().includes('August 23')).toBe(true)
    })

    it('should display slots Of Selected Date', () => {
      const slotList = wrapper.find('.tab')
      expect(slotList.length).toBe(24)
    })

    it('should expect 1 Not Available slots from data', () => {
      const notAvailable = wrapper.find('.allotted')
      const available = wrapper.find('.tab_label')
      expect(notAvailable.length).toBe(1)
      expect(available.length).toBe(23)
    })

    it('should book a slot from data', () => {
      wrapper.find('#acc-open-12').simulate('click')
      wrapper
        .find('#message-12')
        .simulate('change', { target: { value: 'Hello' } })
      wrapper.find('#submit-12').simulate('click')
      const available = wrapper.find('.tab_label')
      expect(available.length).toBe(22)
      const notAvailable = wrapper.find('.allotted')
      expect(notAvailable.length).toBe(1)
      const booked = wrapper.find('.booked')
      expect(booked.length).toBe(1)
    })
  })
})
