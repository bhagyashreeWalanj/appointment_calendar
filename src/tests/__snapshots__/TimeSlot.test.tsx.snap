// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Time Slot component should display to Time Slot Component 1`] = `
<div
  className="timeslots"
>
  <div
    className="timeslots__header"
  >
    Tuesday, August 23rd, 2022
  </div>
  <div
    className="timeslots__body"
  >
    <input
      className="hide_button"
      id="acc-close"
      name="rd"
      type="radio"
    />
    <div
      className="tabs"
    >
      <div
        className="tab"
        key="00:00"
        onClick={[Function]}
      >
        <input
          id="chk-0"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        00:00
        <label
          className="tab_label"
          htmlFor="chk-0"
          id="acc-open-0"
        >
          <span>
            00:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-0"
            name="message_0"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-0"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_0"
        />
      </div>
      <div
        className="tab"
        key="01:00"
        onClick={[Function]}
      >
        <input
          id="chk-1"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        01:00
        <label
          className="tab_label"
          htmlFor="chk-1"
          id="acc-open-1"
        >
          <span>
            01:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-1"
            name="message_1"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-1"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_1"
        />
      </div>
      <div
        className="tab"
        key="02:00"
        onClick={[Function]}
      >
        <input
          id="chk-2"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        02:00
        <label
          className="tab_label"
          htmlFor="chk-2"
          id="acc-open-2"
        >
          <span>
            02:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-2"
            name="message_2"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-2"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_2"
        />
      </div>
      <div
        className="tab"
        key="03:00"
        onClick={[Function]}
      >
        <input
          id="chk-3"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        03:00
        <label
          className="tab_label"
          htmlFor="chk-3"
          id="acc-open-3"
        >
          <span>
            03:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-3"
            name="message_3"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-3"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_3"
        />
      </div>
      <div
        className="tab"
        key="04:00"
        onClick={[Function]}
      >
        <input
          id="chk-4"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        04:00
        <label
          className="tab_label"
          htmlFor="chk-4"
          id="acc-open-4"
        >
          <span>
            04:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-4"
            name="message_4"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-4"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_4"
        />
      </div>
      <div
        className="tab"
        key="05:00"
        onClick={[Function]}
      >
        <input
          id="chk-5"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        05:00
        <label
          className="tab_label"
          htmlFor="chk-5"
          id="acc-open-5"
        >
          <span>
            05:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-5"
            name="message_5"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-5"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_5"
        />
      </div>
      <div
        className="tab"
        key="06:00"
        onClick={[Function]}
      >
        <input
          id="chk-6"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        06:00
        <label
          className="tab_label"
          htmlFor="chk-6"
          id="acc-open-6"
        >
          <span>
            06:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-6"
            name="message_6"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-6"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_6"
        />
      </div>
      <div
        className="tab"
        key="07:00"
        onClick={[Function]}
      >
        <input
          id="chk-7"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        07:00
        <label
          className="tab_label"
          htmlFor="chk-7"
          id="acc-open-7"
        >
          <span>
            07:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-7"
            name="message_7"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-7"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_7"
        />
      </div>
      <div
        className="tab"
        key="08:00"
        onClick={[Function]}
      >
        <input
          id="chk-8"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        08:00
        <label
          className="tab_label"
          htmlFor="chk-8"
          id="acc-open-8"
        >
          <span>
            08:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-8"
            name="message_8"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-8"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_8"
        />
      </div>
      <div
        className="tab"
        key="09:00"
        onClick={[Function]}
      >
        <input
          id="chk-9"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        09:00
        <label
          className="tab_label"
          htmlFor="chk-9"
          id="acc-open-9"
        >
          <span>
            09:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-9"
            name="message_9"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-9"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_9"
        />
      </div>
      <div
        className="tab"
        key="10:00"
        onClick={[Function]}
      >
        <input
          id="chk-10"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        10:00
        <label
          className="tab_label"
          htmlFor="chk-10"
          id="acc-open-10"
        >
          <span>
            10:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-10"
            name="message_10"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-10"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_10"
        />
      </div>
      <div
        className="tab"
        key="11:00"
        onClick={[Function]}
      >
        <input
          id="chk-11"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        11:00
        <label
          className="tab_label"
          htmlFor="chk-11"
          id="acc-open-11"
        >
          <span>
            11:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-11"
            name="message_11"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-11"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_11"
        />
      </div>
      <div
        className="tab"
        key="12:00"
        onClick={[Function]}
      >
        <input
          id="chk-12"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        12:00
        <label
          className="tab_label"
          htmlFor="chk-12"
          id="acc-open-12"
        >
          <span>
            12:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-12"
            name="message_12"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-12"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_12"
        />
      </div>
      <div
        className="tab"
        key="13:00"
        onClick={[Function]}
      >
        <input
          id="chk-13"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        13:00
        <label
          className="tab_label"
          htmlFor="chk-13"
          id="acc-open-13"
        >
          <span>
            13:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-13"
            name="message_13"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-13"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_13"
        />
      </div>
      <div
        className="tab"
        key="14:00"
        onClick={[Function]}
      >
        <input
          id="chk-14"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        14:00
        <label
          className="tab_label"
          htmlFor="chk-14"
          id="acc-open-14"
        >
          <span>
            14:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-14"
            name="message_14"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-14"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_14"
        />
      </div>
      <div
        className="tab"
        key="15:00"
        onClick={[Function]}
      >
        <input
          id="chk-15"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        15:00
        <label
          className="tab_label"
          htmlFor="chk-15"
          id="acc-open-15"
        >
          <span>
            15:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-15"
            name="message_15"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-15"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_15"
        />
      </div>
      <div
        className="tab"
        key="16:00"
        onClick={[Function]}
      >
        <input
          id="chk-16"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        16:00
        <label
          className="tab_label"
          htmlFor="chk-16"
          id="acc-open-16"
        >
          <span>
            16:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-16"
            name="message_16"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-16"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_16"
        />
      </div>
      <div
        className="tab"
        key="17:00"
        onClick={[Function]}
      >
        <input
          id="chk-17"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        17:00
        <label
          className="tab_label"
          htmlFor="chk-17"
          id="acc-open-17"
        >
          <span>
            17:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-17"
            name="message_17"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-17"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_17"
        />
      </div>
      <div
        className="tab"
        key="18:00"
        onClick={[Function]}
      >
        <input
          id="chk-18"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        18:00
        <label
          className="tab_label"
          htmlFor="chk-18"
          id="acc-open-18"
        >
          <span>
            18:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-18"
            name="message_18"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-18"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_18"
        />
      </div>
      <div
        className="tab"
        key="19:00"
        onClick={[Function]}
      >
        <input
          id="chk-19"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        19:00
        <label
          className="tab_label"
          htmlFor="chk-19"
          id="acc-open-19"
        >
          <span>
            19:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-19"
            name="message_19"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-19"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_19"
        />
      </div>
      <div
        className="tab"
        key="20:00"
        onClick={[Function]}
      >
        <input
          id="chk-20"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        20:00
        <label
          className="tab_label"
          htmlFor="chk-20"
          id="acc-open-20"
        >
          <span>
            20:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-20"
            name="message_20"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-20"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_20"
        />
      </div>
      <div
        className="tab"
        key="21:00"
        onClick={[Function]}
      >
        <input
          id="chk-21"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        21:00
        <label
          className="allotted"
          htmlFor=""
          id="acc-open-21"
        >
          <span>
            21:00
          </span>
          Not Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-21"
            name="message_21"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-21"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_21"
        />
      </div>
      <div
        className="tab"
        key="22:00"
        onClick={[Function]}
      >
        <input
          id="chk-22"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        22:00
        <label
          className="tab_label"
          htmlFor="chk-22"
          id="acc-open-22"
        >
          <span>
            22:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-22"
            name="message_22"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-22"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_22"
        />
      </div>
      <div
        className="tab"
        key="23:00"
        onClick={[Function]}
      >
        <input
          id="chk-23"
          name="rd"
          style={
            Object {
              "display": "none",
            }
          }
          type="radio"
        />
        23:00
        <label
          className="tab_label"
          htmlFor="chk-23"
          id="acc-open-23"
        >
          <span>
            23:00
          </span>
          Available
        </label>
        <div
          className="tab_content"
        >
          <h4>
            Reason for a Call :
          </h4>
          <input
            className="textBox"
            id="message-23"
            name="message_23"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="confirmButton"
            id="submit-23"
            onClick={[Function]}
            type="submit"
          >
            Confirm
          </button>
        </div>
        <label
          className="box_close"
          htmlFor="acc-close"
          id="close_23"
        />
      </div>
    </div>
  </div>
</div>
`;
